src/index.ts

```diff
@@ -0,0 +1,75 @@
+import fs from "fs";
+import path from "path";
+import semver from "semver";
+
+const VERSION_NUMBER = /^v?\d+$/;
+
+async function main() {
+    let targetVersionArg = process.argv[2];
+
+    if (targetVersionArg === undefined) {
+        console.error("Missing target version! Usage: npx @comet/upgrade <version>");
+        process.exit(-1);
+    }
+
+    if (!VERSION_NUMBER.test(targetVersionArg)) {
+        console.error("Can't parse version number. Example usage: npx @comet/upgrade v4");
+        process.exit(-1);
+    }
+
+    if (targetVersionArg.startsWith("v")) {
+        targetVersionArg = targetVersionArg.substring(1);
+    }
+
+    const targetVersion = Number(targetVersionArg);
+
+    const scriptsFolder = path.join(__dirname, `v${targetVersion}`);
+
+    if (!fs.existsSync(scriptsFolder)) {
+        console.error(`Can't find target version 'v${targetVersionArg}'`);
+        listTargetVersions();
+        process.exit(-1);
+    }
+
+    const currentVersion = getCurrentVersion();
+
+    console.info(`Upgrading from v${currentVersion} to v${targetVersion}`);
+    await runUpgradeScripts(targetVersion);
+}
+
+function getCurrentVersion() {
+    if (!fs.existsSync("admin/node_modules/@comet/admin/package.json")) {
```

Comment:
@nsams is it sufficient to only check the version of the @comet/admin package? Should we consider a case where a project only has an API, or API + site? Also, would you check the installed version or the project's package.json? Checking the installed version is easier, but requires the project to be installed. Checking the version in the package.json might be tricky due to range descriptors, e.g. `~3.0.0`, `>=3.0.0`

Reply:
> is it sufficient to only check the version of the @comet/admin package? Should we consider a case where a project only has an API, or API + site?

I think it is ok if we cover only 95% if that simplifies things. Improvements can always be made later on. (you might run into issues once we merge @comet/admin into @comet/admin-cms - but you can fix that then - once we know for sure how everything is named)

> also, would you check the installed version or the project's package.json?

yes, don't expect everything installed properly, when upgrading multiple versions no install might be done in between.

> Checking the version in the package.json might be tricky due to range descriptors, e.g. ~3.0.0, >=3.0.0

It is ok to support only our commonly used syntax (95%)

Reply:
I've decided to stick with only checking for the `@comet/admin` package for now, as it simplifies things.

> yes, don't expect everything installed properly, when upgrading multiple versions no install might be done in between.

Changed it to check the package.json

> It is ok to support only our commonly used syntax (95%)

I now extract the major.minor.patch syntax with a regex. Should work for most cases, but fails for version ranges such as `*` or `canary`. We don't use them in our projects, so it's okay